esmvaltool/preprocessor/_area.py

```diff
@@ -179,14 +204,21 @@ def average_region(cube, coord1, coord2, fx_files=None):
         logger.info('Calculated grid area:{}'.format(grid_areas.shape))
 
     if cube.data.shape != grid_areas.shape:
-
         raise ValueError('Cube shape ({}) doesn`t match grid area shape '
                          '({})'.format(cube.data.shape, grid_areas.shape))
 
-    result = cube.collapsed([coord1, coord2],
-                            iris.analysis.MEAN,
-                            weights=grid_areas)
-    return result
+    operation = get_iris_analysis_operation(operator)
+
+    # TODO: implement weighted stdev, median, and var when available in iris.
+    # See iris issue: https://github.com/SciTools/iris/issues/3208
+
+    if operator in ['mean', ]:
```

Comment:
isn't `average_region` supposed to use `mean` as collapsing statistic at all times?

Reply:
you could create a more general function like `aggregated_region` that could work with more than one collapsing statistic, but in this case the call to `get_iris_analysis_operation` is probably overkill

Reply:
I don't think that this functionality is unwarranted; I can image plenty of scenarios where these functions would be useful. However, I do agree that it's unfortunate to put it inside a function which is already named as `average_region`. The default behaviour for `average_region` remains as an area weighted mean.

Also, I'd like to point out that the function `zonal_means` has had this ability the entire time. I simply moved code from `zonal_means` into a new generic function `get_iris_analysis_operation` that was called by `zonal_means` and `average_region`.